### ISSUE TITLE
mismtach between document and implementation

### ISSUE THREAD

**Issue by lifengjin** [Lightning-AI/metrics#1366]
https://github.com/Lightning-AI/metrics/blob/2a055f5594a624685e26ba64bf20ab0d12225c86/src/torchmetrics/functional/text/perplexity.py#L86

The document says pred is probabilities, but softmax is applied to pred again treating it like logits.

**Comment by Borda** [comment]
@lifengjin mind sending a frat with a suggested fix? :rabbit:

**Comment by lifengjin** [comment]
what is a frat?

**Comment by SkafteNicki** [comment]
@lifengjin I think its just a type, should probably be PR :]
Feel free to send one, else I am going to do it within one of the next days.
I think the correct fix here is to fix the documentation such that it says log-probabilities instead of probabilities.

**Comment by lifengjin** [comment]
@SkafteNicki Please go ahead! Thanks.
